paddle/fluid/pir/dialect/operator/ir/ops.yaml

```
@@ -1163,6 +1163,15 @@
   backward : pad_grad
   interfaces : paddle::dialect::InferSymbolicShapeInterface
 
+- op : partial_allgather
+  args : (Tensor x, int nranks, int rank, int ring_id = 0, bool use_calc_stream = false)
+  output : Tensor(out)
+  infer_meta :
+    func: PartialAllgatherInferMeta
+  kernel :
+    func : partial_allgather
+    data_type : dtype
```

```suggestion
    data_type : dtype
```
为什么要添加这个呢？

另外，这个Op声明了PartialAllGatherOpInplaceInferer，所以需要额外声明inplace字段
```
  inplace: (x->out)
```

---

data_type这里我复制错了，感谢纠正～